smriprep/cli/run.py

```diff
@@ -176,8 +176,16 @@ def _warn_redirect(message, category, filename, lineno, file=None, line=None):
     warnings.showwarning = _warn_redirect
 
     # Precedence: --fs-license-file, $FS_LICENSE, default_license
-    license_file = opts.fs_license_file or os.getenv('FS_LICENSE')
-    if not check_valid_fs_license(lic=license_file):
+    fs_home = os.getenv("FREESURFER_HOME")
+    license_file = (
+        opts.fs_license_file
+        or os.getenv('FS_LICENSE')
+        or os.path.join(fs_home, "license.txt") if fs_home else None
+    )
+    if os.path.isfile(license_file):
+        os.environ["FS_LICENSE"] = license_file
```

Comment:
Note that the operator precedence makes this

```Python
(opts.fs_license_file
 or os.getenv('FS_LICENSE')
 or os.path.join(fs_home, "license.txt")
) if fs_home else None
```

I think this is fine, because `fs_home` has to be defined for FreeSurfer to work at all, so `license_file` will always be a string, but if it were None `os.path.isfile` would fail.

But I think the logic should be:

```suggestion
    if opts.fs_license_file is not None:
        os.environ["FS_LICENSE"] = opts.fs_license_file
```

`check_valid_fs_license` will find `$FS_LICENSE` or `$FREESURFER_HOME/license.txt`, so we don't need to modify the environment except in this one case.

Reply:
This is exactly the point, `check_valid_fs_license` needs to be operated assuming it is ignorant of the environment. It needs just to run `mri_convert` and inform out about the result.

The most effective way of making multiproc workers to be able to run FS is just setting FS_LICENSE.

Reply:
Addressed the problem of precedence (i think)

Reply:
The code I suggested allows check_valid_fs_license to be ignorant of the environment.

Reply:
Sorry, I misunderstood your comments. I just pushed a commit with your suggestion.